test/test_torrent.cpp

```diff
@@ -52,110 +52,117 @@ POSSIBILITY OF SUCH DAMAGE.
 using namespace libtorrent;
 namespace lt = libtorrent;
 
-void test_running_torrent(boost::shared_ptr<torrent_info> info, boost::int64_t file_size)
-{
-	settings_pack pack = settings();
-	pack.set_int(settings_pack::alert_mask, alert::storage_notification);
-	pack.set_str(settings_pack::listen_interfaces, "0.0.0.0:48130");
-	pack.set_int(settings_pack::max_retry_port_bind, 10);
-	lt::session ses(pack);
+namespace {
+	bool wait_priority(torrent_handle const& h, std::vector<int> const& prio)
+	{
+		for (int i = 0; i < 10; ++i)
+		{
+			if (h.file_priorities() == prio) { return true; }
 
-	std::vector<boost::uint8_t> zeroes;
-	zeroes.resize(1000, 0);
-	add_torrent_params p;
-	p.flags &= ~add_torrent_params::flag_paused;
-	p.flags &= ~add_torrent_params::flag_auto_managed;
-	p.ti = info;
-	p.save_path = ".";
+			test_sleep(100);
```

Comment:
ideally this should be waiting for an alert. but I'm not sure it's possible to add a new alert without risking ABI incompatibilities. maybe it has to be added to `master`

Reply:
Good idea, but for now this should stay, isn't it?

Reply:
yeah